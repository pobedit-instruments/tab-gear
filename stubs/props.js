import { bindActionCreators } from 'redux';

import actual from './tabs';
import * as Actions from '../views/actions';
import * as ActionTypes from '../views/constants/ActionTypes';
import config from '../config';

export const tabs = {
	actual,
	masked: false,
	active: actual.find(tab => tab.active),
	search: actual
};

export const type = ActionTypes.SHOW_TABS;

export const actions = new function () {
	let actions = {};

	for (let [name, value] of Object.entries(Actions)) {
		actions[name] = bindActionCreators(value, () => {});
	}

	return actions;
};

export { config };
