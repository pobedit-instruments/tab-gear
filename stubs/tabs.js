export default [
	{
		title : 'Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform',
		url   : 'https://cloud.google.com',
		favIconUrl: 'https://cloud.google.com/_static/cc7442d8c7' +
					'/images/cloud/icons/favicons/favicon.ico',
		id : 0,
		index : 0,
		active: true,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Plus +++++++++++++++++++++++++++++++++++++++++++',
		url   : 'https://developers.google.com/+/',
		favIconUrl: 'https://developers.google.com/site-assets/logo-google-plus-white.svg',
		id : 1,
		index : 1,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Play',
		url   : 'https://play.google.com',
		favIconUrl: 'https://developers.google.com/_static/cc7442d8c7/images/favicon.png',
		id : 2,
		index : 2,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Maps',
		url   : 'https://maps.google.com',
		favIconUrl: 'https://www.google.com/images/branding/product/ico/maps_32dp.ico',
		id : 3,
		index : 3,
		active: false,
		highlighted: false,
		incognito: true
	},

	{
		title : 'Google Translate',
		url   : 'https://translate.google.ru/',
		favIconUrl: 'https://translate.google.ru/favicon.ico',
		id : 4,
		index : 4,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Android',
		url   : 'https://developer.android.com',
		favIconUrl: 'https://developer.android.com/favicon.ico',
		id : 5,
		index : 5,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Chrome',
		url   : 'https://google.com/chrome',
		favIconUrl: 'https://www.google.com/images/icons/product/chrome-32.png',
		id : 6,
		index : 6,
		highlighted: true,
		active: false
	},

	{
		title : 'Google Developers',
		url   : 'https://developers.google.com/web/',
		favIconUrl: 'https://developers.google.com/web/images/web-fundamentals-icon192x192.png',
		id : 7,
		index : 7,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Firebase',
		url   : 'https://firebase.google.com/',
		favIconUrl: 'https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png',
		id : 9,
		index : 9,
		highlighted: false,
		active: false
	}
];
