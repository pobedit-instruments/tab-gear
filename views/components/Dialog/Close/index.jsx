import React, { Component, PropTypes } from 'react';
import { bind } from 'decko';

class Close extends Component {
	constructor (properties) {
		super(...arguments);
	}

	/**
	 * Close all tabs
	 *
	 * @param {Event} event
	 */
	@bind
	closeAllTabs (event) {
		let { tabs, actions } = this.props;
		let { actual } = tabs;

		actions.Layout.closeAllTabs(actual);
		actions.Tabs.showTabs(actual);

		event.preventDefault();
	}

	/**
	 * Close all tabs
	 *
	 * @param {Event} event
	 */
	@bind
	ignoreTabDialog (event) {
		window.localStorage.setItem('tg-option-dialog', event.target.checked);
	}

	render () {
		return <div>
					<p className="tg-block">
						{ chrome.i18n.getMessage('attention_closing') }
					</p>

					<button className="tg-button" onClick={ this.closeAllTabs } autoFocus="autoFocus">
						{ chrome.i18n.getMessage('confirm') }
					</button>

					<p className="tg-block">
						<input type="checkbox" id="confirm" onChange={ this.ignoreTabDialog } />
						<label htmlFor="confirm" className="tg-label">
							{ chrome.i18n.getMessage('prevent_confirmation') }
						</label>
					</p>
				</div>;
	}
}

Close.propTypes = {
	tabs   : PropTypes.object.isRequired,
	actions: PropTypes.object.isRequired
};

export default Close;
