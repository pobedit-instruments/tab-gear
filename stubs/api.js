import chrome from 'chrome-stub';

import tabs from './tabs';
import locale from '../locale/en/messages.json';

chrome.i18n._locales = locale;

let methods = [
	'query',
	'remove',
	'move',
	'create',
	'discard',
	'highlight'
];

for (let method of methods) {
	chrome.tabs[method].yields(tabs);
}

export default chrome;
