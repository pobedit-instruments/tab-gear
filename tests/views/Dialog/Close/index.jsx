import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import Close from '../../../../views/components/Dialog/Close';
import { tabs, actions } from '../../../../stubs/props';

test('Dialog::Close view', () => {
	let component = renderer.create(<Close tabs={ tabs } actions={ actions } />);
	let view = component.toJSON();

	let actual = expect(view);

	actual.toMatchSnapshot();
});
