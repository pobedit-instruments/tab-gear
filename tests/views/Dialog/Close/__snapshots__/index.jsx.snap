exports[`test Dialog::Close view 1`] = `
<div>
  <p
    className="tg-block">
    Are you sure you want to close all tabs?
  </p>
  <button
    autoFocus="autoFocus"
    className="tg-button"
    onClick={[Function]}>
    Confirm
  </button>
  <p
    className="tg-block">
    <input
      id="confirm"
      onChange={[Function]}
      type="checkbox" />
    <label
      className="tg-label"
      htmlFor="confirm">
      Prevent additional confirmation
    </label>
  </p>
</div>
`;
